packages/pluggableWidgets/signature-native/src/Signature.tsx

```diff
@@ -0,0 +1,115 @@
+import { flattenStyles } from "@native-mobile-resources/util-widgets";
+import { executeAction } from "@widgets-resources/piw-utils";
+import { createElement, ReactElement, useRef } from "react";
+import { View, Text } from "react-native";
+import SignatureScreen, { SignatureViewRef } from "react-native-signature-canvas";
+import { extractStyles } from "../../../tools/util-widgets/src/styles";
+import { Touchable } from "../components/Touchable";
+
+import { SignatureProps } from "../typings/SignatureProps";
+import { SignatureStyle, defaultSignatureStyle } from "./ui/Styles";
+
+export type Props = SignatureProps<SignatureStyle>;
+
+export function Signature(props: Props): ReactElement {
+    const ref = useRef<SignatureViewRef>(null);
+    const styles = flattenStyles(defaultSignatureStyle, props.style);
+    const [signatureProps, containerStyles] = extractStyles(styles.container, ["penColor", "backgroundColor"]);
+    const [buttonClearContainerProps, buttonClearContainerStyles] = extractStyles(styles.buttonClearContainer, [
+        "rippleColor",
+        "activeOpacity",
+        "underlayColor"
+    ]);
+    const [buttonSaveContainerProps, buttonSaveContainerStyles] = extractStyles(styles.buttonSaveContainer, [
+        "rippleColor",
+        "activeOpacity",
+        "underlayColor"
+    ]);
+    const buttonCaptionClear =
+        props.buttonCaptionClear && props.buttonCaptionClear.value ? props.buttonCaptionClear.value : "Clear";
+    const buttonCaptionSave =
+        props.buttonCaptionSave && props.buttonCaptionSave.value ? props.buttonCaptionSave.value : "Save";
+    const webStyles = `
+                    .m-signature-pad {
+                        border: none;
+                    }
+                    .m-signature-pad--body {
+                      left: 0;
+                      right: 0;
+                      top: 0;
+                      bottom: 0;
+                      border: none;
+                    }
+                    .m-signature-pad--body canvas {
+                        border-radius: 0;
+                        box-shadow: none;
+                    }
+                    .m-signature-pad--footer {
+                        display: none;
+                    }
+                `;
+
+    const handleSignature = (base64signature: string): void => {
```

Comment:
I think this deserves a `useCallback` with `props.imageAttribute` and `props.onSave` as dependency